src/strategy_manager/strategy_manager.cpp

```diff
@@ -0,0 +1,156 @@
+/******************************************************************************
+ * Copyright © 2013-2019 The Komodo Platform Developers.                      *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * Komodo Platform software, including this file may be copied, modified,     *
+ * propagated or distributed except according to the terms contained in the   *
+ * LICENSE file                                                               *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <vector>
+#include <unordered_map>
+
+#include "strategy_manager.hpp"
+
+namespace antara {
+  bool asset::operator==(const asset &other) const {
+    return symbol == other.symbol;
+  };
+
+  std::pair<antara::asset, antara::asset> pair::to_std_pair() {
+    return std::pair<antara::asset, antara::asset>(base, quote);
+  };
+
+  bool pair::operator==(const pair &rhs) const {
+    return base == rhs.base && quote == rhs.quote;
+  };
+}
+
+namespace std {
+
+  template <>
+  struct hash<antara::pair> {
+    std::size_t operator()(const antara::pair& p) const {
+      using std::size_t;
+      using std::hash;
+      using std::string;
+
+      std::size_t h1 = std::hash<std::string>{}(p.base.symbol);
+      std::size_t h2 = std::hash<std::string>{}(p.quote.symbol);
+
+      return h1 ^ (h2 << 1);
+    }
+  };
+}
+
+enum class Side { BUY, SELL, BOTH };
+
+class MMStrategy {
+public:
+  antara::pair pair;
+  float spread;
+  float quantity;
+  Side side;
+};
+
+class OrderLevel {
+public:
+  float price;
+  float quantity;
+  Side side;
+
+  OrderLevel(float price, float quantity, Side side) {
+    this->price = price;
+    this->quantity = quantity;
+    this->side = side;
+  }
+};
+
+class OrderSet {
+public:
+  antara::pair pair;
+  std::vector<OrderLevel> levels;
+
+  OrderSet();
+
+  OrderSet(antara::pair pair, std::vector<OrderLevel> levels) {
```

Comment:
in c++ we use reference for not copying each object https://www.ntu.edu.sg/home/ehchua/programming/cpp/cp4_pointerreference.html

Reply:
all your field should be private since you use it inside the class